File: Bachelor/Models/Communities/Post.cs
using Bachelor.Models.Communities;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Bachelor.Models.Communities
{
    public class Post
    {
        [Key]
        public int Id { get; set; }

        [RegularExpression(@"[a-zA-ZæøåÆØÅ., \-\s\S]{3,1000}$")]
        public string Text { get; set; }

        public virtual User User { get; set; }

        public virtual Community Community { get; set; }

        public string Date { get; set; }

        [RegularExpression(@"^-?\d+$")]
        public int Upvotes { get; set; }

        [RegularExpression(@"^-?\d+$")]
        public int Downvotes { get; set; }
        public virtual List<Comment> Comment { get; set; }

        public virtual PostTag PostTag { get; set; }

        public bool Anonymous { get; set; }
    }
}

File: Bachelor/ClientApp/src/app/Models/Communities/Post.ts
import { Experience } from "../Users/Experience";
import { User } from "../Users/User";
import { Comment } from "./Comment";
import { Community } from "./Community";
import { PostTag } from "./PostTag";

export class Post {
  id: number;
  text: string;
  user: User;
  community: Community;
  date: string;
  upvotes: number;
  downvotes: number;
  comment: Comment[];
  postTag: PostTag;
  anonymous: boolean;
  experience: Experience;
} 
